crates/ruff_linter/src/noqa.rs

```diff
@@ -424,21 +425,23 @@ impl<'a> FileNoqaDirectives<'a> {
 /// An individual file-level exemption (e.g., `# ruff: noqa` or `# ruff: noqa: F401, F841`). Like
 /// [`FileNoqaDirectives`], but only for a single line, as opposed to an aggregated set of exemptions
 /// across a source file.
-#[derive(Debug, PartialEq, Eq)]
+#[derive(Debug)]
 pub(crate) enum ParsedFileExemption<'a> {
     /// The file-level exemption ignores all rules (e.g., `# ruff: noqa`).
     All,
     /// The file-level exemption ignores specific rules (e.g., `# ruff: noqa: F401, F841`).
-    Codes(Vec<&'a str>),
+    Codes(Codes<'a>),
 }
 
 impl<'a> ParsedFileExemption<'a> {
     /// Return a [`ParsedFileExemption`] for a given comment line.
-    fn try_extract(line: &'a str) -> Result<Option<Self>, ParseError> {
-        let line = Self::lex_whitespace(line);
+    fn try_extract(comment_range: TextSize, source: &'a str) -> Result<Option<Self>, ParseError> {
```

Comment:
Sorry that I haven't been clear about what signature I'd expect. I think this signature still has the same "flaw" in that it's unclear if `comment_range` is an offset relative to `source` or the entire source text of the file. 

To avoid this, I suggest changing the signature to `try_extract(comment_range: TextRange, source; &'a str)` and then call it like this: `ParsedFileExemption::try_extract(range, locator.contents())`; It's then the `try_extract`'s responsible to slice the comment text out of the source text.

Reply:
Oh sorry about that. I missed that you suggested `TextRange` instead of `TextSize` :facepalm: This makes much more sense to me now.